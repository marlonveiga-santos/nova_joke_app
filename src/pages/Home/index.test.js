import {render} from "@testing-library/react";
import Home from ".";

/* Spy Jest que emula o DOM para aceitar a função scrollTo */
global.scrollTo = jest.fn();

test("Renders Home snapshot", () => {
  /* Escreve o componente Home atual como um React.Fragment permitindo a comparação com snapshot */
  const {asFragment} = render(<Home />);

  /* Compara o fragmento com a snapshot existente */
  expect(asFragment(<Home />)).toMatchSnapshot();
});