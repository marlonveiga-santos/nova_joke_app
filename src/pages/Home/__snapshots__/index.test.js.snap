// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Home snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <h1>
      Welcome to Nova Joke APP
    </h1>
  </header>
  <div
    class="loading"
  >
    Loading...
    <p>
      If persists check your network or refresh the page
    </p>
  </div>
  <footer>
    Â© 2021 Marlon Veiga
    <a
      href="https://github.com/marlonveiga-santos"
    >
      <h3>
        See more in Github
      </h3>
    </a>
  </footer>
</DocumentFragment>
`;
